syncer/syncer.go

```diff
@@ -74,6 +78,11 @@ func (s *syncer) initializePeerMap() {
 
 	for peerStatus := range s.syncPeerClient.GetPeerStatusUpdateCh() {
 		s.peerMap.Put(peerStatus)
+
+		select {
+		case s.newStatusCh <- struct{}{}:
+		default:
+		}
```

Comment:
I think you can emit outside loop

Reply:
I think we can not put it outsthe ide loop. Loop will be trigger on every new status, so once we put status in peerMap we need to emit an event.